public-ui/src/app/data-browser/views/survey-chart/survey-chart.component.ts

```diff
@@ -64,26 +75,31 @@ export class SurveyChartComponent {
         q.selectedAnalysis = q.genderAnalysis;
         break;
     }
-    this.selectGraph(q.graphDataToShow, q, answer);
-  }
-
-  public selectGraph(sg: any, q: any, answer: any) {
-    q.graphDataToShow = sg;
+    this.selectedChartAnalysis = q.selectedAnalysis;
+    this.displayGraphErrorMessage = q.selectedAnalysis === undefined;
+    // sends information to google analyitics
     this.dbc.triggerEvent('graphTabClick', 'Survey Graph',
       'Click', this.surveyName + ' - ' + q.graphToShow + ' - Q'
       + q.actualQuestionNumber + ' - ' + q.conceptName + ' - ' + answer.stratum4 +
       ' - ' + this.graphToShow, this.searchTerm, null);
-    switch (q.graphToShow) {
-      case GraphType.BiologicalSex:
-        q.selectedAnalysis = q.genderAnalysis;
-        break;
-      case GraphType.AgeWhenSurveyWasTaken:
-        q.selectedAnalysis = q.ageAnalysis;
-        break;
-    }
-    this.displayGraphErrorMessage = q.selectedAnalysis === undefined;
+    // this.selectGraph(q.graphDataToShow, q, answer);
   }
 
+  /// ----<<<< this code seem to be redundent >>>>>>--------
+  // public selectGraph(sg: any, q: any, answer: any) {
+  //   q.graphDataToShow = sg;
+
+  //   switch (q.graphToShow) {
+  //     case GraphType.BiologicalSex:
+  //       q.selectedAnalysis = q.genderAnalysis;
+  //       break;
+  //     case GraphType.AgeWhenSurveyWasTaken:
+  //       q.selectedAnalysis = q.ageAnalysis;
+  //       break;
+  //   }
+  //   this.displayGraphErrorMessage = q.selectedAnalysis === undefined;
+  // }
+
```

Comment:
I am going to remove this code. It seemed to be redundant. I moved the displayGraphErrorMessage to the function above. @SrushtiGangireddy  do you know why it might be necessary?